designcompose/src/main/java/com/android/designcompose/FrameRender.kt

```diff
@@ -229,6 +233,35 @@ private fun calculateArcData(
     return returnShape
 }
 
+// Set up the paint object to render a vector path as a stroke with a single dash that matches the
+// length of the current progress within the vector.
+private fun calculateProgressVectorData(
+    data: ProgressVectorMeterData,
+    paths: List<Path>,
+    p: Paint,
+    style: ViewStyle,
+    meterValue: Float,
+    density: Float
+) {
+    val strokeWidth = style.node_style.stroke.stroke_weight.toUniform() * density
+    val discretizedMeterValue = meterValue.coerceDiscrete(data.discrete, data.discreteValue)
+
+    // Get full length of path
+    var pathLen = 0f
+    paths.forEach {
```

Comment:
I also wonder if this is something that can be done during doc conversion, but perhaps there are scenarios where layout can change the path length?

Reply:
I think scaled vectors can cause layout to change the size. We don't support scaled vector paths properly right now, though we should.